webapp/src/components/SingleStudy/MapView/index.tsx

```diff
@@ -0,0 +1,391 @@
+          const links = linkData.filter((o) => o.source !== id || o.target !== target);
+          setLinkData(links);
+          setSelectedItem(undefined);
+          await deleteLink(study.id, id, target);
+        } catch (e) {
+          setLinkData([...linkData]);
+          enqueueErrorSnackbar(enqueueSnackbar, t('singlestudy:deleteAreaOrLink'), e as AxiosError);
+        }
+      } else if (nodeData && linkData && !target) {
+        const obj = nodeData.filter((o) => o.id !== id);
+        const links = linkData.filter((o) => o.source !== id && o.target !== id);
+        try {
+          setSelectedItem(undefined);
+          setLinkData(links);
+          setNodeData(obj);
+          await deleteArea(study.id, id);
+        } catch (e) {
+          setLinkData([...linkData]);
+          setNodeData([...nodeData]);
+          enqueueErrorSnackbar(enqueueSnackbar, t('singlestudy:deleteAreaOrLink'), e as AxiosError);
+        }
+      }
+    }, 0);
+  };
+
+  useEffect(() => {
+    const init = async () => {
+      if (firstNode && secondNode) {
+        try {
+          setLinkData([...linkData, ...[{
+            source: firstNode,
+            target: secondNode,
+          }]]);
+          setFirstNode(undefined);
+          setSecondNode(undefined);
+          await createLink(study.id, { area1: firstNode, area2: secondNode });
+        } catch (e) {
+          setLinkData(linkData.filter((o) => o.source !== firstNode || o.target !== secondNode));
+          enqueueErrorSnackbar(enqueueSnackbar, t('singlestudy:createLinkError'), e as AxiosError);
+        }
+      }
+    };
+    init();
+  }, [enqueueSnackbar, t, firstNode, secondNode, study.id, linkData]);
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const data = await getSynthesis(study.id);
+        setLinkData(Object.keys(data.areas).reduce((links, currentAreaId) =>
+          links.concat(Object.keys(data.areas[currentAreaId].links).map((linkId) => ({
+            source: currentAreaId,
+            target: linkId,
+          }))), [] as Array<LinkProperties>));
+        const areaData = await getAreaPositions(study.id, Object.keys(data.areas).join(','));
+        if (Object.keys(data.areas).length === 1) {
+          const area = { [Object.keys(data.areas)[0]]: areaData as SingleAreaConfig };
+          const tempNodeData = Object.keys(area).map((areaId) => ({
+            id: areaId,
+            x: area[areaId].ui.x,
+            y: area[areaId].ui.y,
+            color: `rgb(${area[areaId].ui.color_r}, ${area[areaId].ui.color_g}, ${area[areaId].ui.color_b})`,
+            rgbColor: [area[areaId].ui.color_r, area[areaId].ui.color_g, area[areaId].ui.color_b],
+            size: { width: calculateSize(areaId), height: NODE_HEIGHT },
+          }));
+          setNodeData(tempNodeData);
+        } else {
+          const area = areaData as AreasConfig;
+          const tempNodeData = Object.keys(area).map((areaId) => ({
+            id: areaId,
```

Comment:
add a "name" properties, the name should be retrieved from data.areas[areaId].name